packages/@react-spectrum/overlays/src/Modal.tsx

```diff
@@ -10,33 +10,40 @@
  * governing permissions and limitations under the License.
  */
 
-import {classNames, useDOMRef, useStyleProps} from '@react-spectrum/utils';
-import {DOMProps, DOMRef, StyleProps} from '@react-types/shared';
+import {classNames, filterDOMProps, useDOMRef, useStyleProps} from '@react-spectrum/utils';
+import {DOMRef} from '@react-types/shared';
+import {mergeProps} from '@react-aria/utils';
 import {ModalProps} from '@react-types/overlays';
 import modalStyles from '@adobe/spectrum-css-temp/components/modal/vars.css';
 import {Overlay} from './Overlay';
 import overrideStyles from './overlays.css';
-import React, {forwardRef} from 'react';
+import React, {forwardRef, HTMLAttributes, ReactNode, RefObject} from 'react';
 import {Underlay} from './Underlay';
 import {useModal, useOverlay, usePreventScroll} from '@react-aria/overlays';
 
-interface ModalWrapperProps extends DOMProps, ModalProps, StyleProps {
-  isOpen?: boolean
+interface ModalWrapperProps extends HTMLAttributes<HTMLElement> {
+  children: ReactNode,
+  isOpen?: boolean,
+  onClose?: () => void,
+  type?: 'fullscreen' | 'fullscreenTakeover',
+  isDismissable?: boolean
 }
 
 function Modal(props: ModalProps, ref: DOMRef<HTMLDivElement>) {
   let {children, onClose, type, isDismissable, ...otherProps} = props;
+  let domRef = useDOMRef(ref);
   let {styleProps} = useStyleProps(props);
 
   return (
     <Overlay {...otherProps}>
       <Underlay />
       <ModalWrapper
+        {...filterDOMProps(otherProps)}
```

Comment:
We have otherProps going to both the `Overlay` and `ModalWrapper`. I think this is true for `Popover` and `Tray` too.